Handle null values in dashboard #11529

rowheat02 opened the pull request:
## Description
<!-- A few sentences describing the overall goals of the pull request' s commits. -->

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Enhancement

<!-- add here the ReadTheDocs link (if needed) -->

fixes #11529

**What is the current behavior?**
<!-- You can also link to an existing issue here -->
#11529
- Null values of Group by fields can not be filtered which is making the chart unreadable sometimes as it's value is being used to calculate the height of the chart 

**What is the new behavior?**
<!-- Describe here the new behaviour based on your changes -->
- The user can remove the null value of the group by field. 
- As Null values was not being rendered on chart by plotly so respective placeholder value for null on Group by field can be updated by the field type.
   - e.g: for string : "NULL" by default, user can also change it
   - for number placeholder value should be number
   - Similarly, for all data types, placeholder value is handled accordingly 
   -  <img width="475" height="637" alt="image" src="https://github.com/user-attachments/assets/6e5598b4-47a2-45e2-9cd2-6220580c9f62" />
  


## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications -->

## Other useful information
For test, types layer can be used from gs-stable for all type of data types.
<img width="475" height="571" alt="image" src="https://github.com/user-attachments/assets/4ca7a6b7-23b0-42a7-9222-b9689363bfd4" />

offtherailz reviewed (changes requested):
- [ ] I see that the checkboxes in fact are mutually exclusive, so having checkbox can be redundant and misleading. I think it makes more sense to have a dedicated section at the bottom like this: 

```
Null values handling

Group by attribute:   // or X 
| default      V | // dropdown
------------------
|- exclude       |
| - placeholder  |
```
If placeholder is selected, show the placeholder. If the scrolls hides for some reason, preserve some space, or make it simply disabled.

- [ ] Instead of the variables `excludeNullGroupByFieldValue`, `useNullPlaceholderForGroupByFieldValue`  etc... I'd use a more structured approach

```json
"options": { 
    "nullHandling": {
          "groupByAttributes": {
              "strategy": "include"(default)|"exclude"|"placeholder", 
              "placeholder":  "NULL"
           }
       }
}
```

rowheat02 commented:
<img width="628" height="348" alt="image" src="https://github.com/user-attachments/assets/afc66943-9f01-409b-9db6-6ec60fb80aa4" />

As suggested, updated
- UI
- variable names
- Help pop-up added
- Tests updated and added
